provider/adapters/src/fallback/adapter.rs

```diff
@@ -16,7 +16,12 @@ use crate::helpers::result_is_err_missing_data_options;
 /// use icu_provider::hello_world::*;
 /// use icu_provider_adapters::fallback::LocaleFallbackProvider;
 ///
-/// let provider = icu_testdata::unstable_no_fallback();
+/// # let provider = icu_provider_adapters::fork::ForkByKeyProvider::new(HelloWorldProvider.as_any_provider(), icu_testdata::any());
+/// # let HelloWorldAndFallbackProvider = provider.as_downcasting();
+/// // This data provider contains both the data we want (`HelloWorldV1`),
+/// // as well as the fallback data (`LocaleFallbackLikelySubtagsV1`,
+/// // `LocaleFallbackParentsV1`, `CollationFallbackSupplementV1`).
```

Comment:
Nit: I think this setup is important and it's confusing to hide it in `#` signs, _especially_ since `HelloWorldAndFallbackProvider` is neither something users can use (it's not exported) _and_ it's not even a type; it's a variable

I still prefer keeping HelloWorldV1 in testdata and reverting this change